### PR TITLE
Zap Zap Zap!

### PR DESCRIPTION
**Pull request by ash2k**

### CODE REVIEW

File: cmd/crd/main.go [hunk 1]
```diff
@@ -14,7 +14,8 @@ import (
 
 func main() {
 	if err := innerMain(); err != nil {
-		log.Fatal(err)
+		fmt.Fprintf(os.Stderr, "%#v", err)
```
Comment (nilebox) [inline cmd/crd/main.go]: why don't we use zap everywhere? isn't it initialized yet by this point?

File: cmd/smith/app/app.go [hunk 2]
```diff
@@ -74,13 +74,14 @@ type App struct {
...
 func (a *App) Run(ctx context.Context) error {
+	defer a.Logger.Sync()
 	// Plugins
 	pluginContainers, err := a.loadPlugins()
 	if err != nil {
 		return err
 	}
 	for pluginName := range pluginContainers {
-		log.Printf("Loaded plugin: %q", pluginName)
+		a.Logger.Sugar().Infof("Loaded plugin: %q", pluginName)
```
Comment (nilebox) [inline cmd/smith/app/app.go]: every call on `.Sugar()` creates a new instance of logger, so probably to invoke it only once.
See https://github.com/uber-go/zap/blob/master/logger.go#L126
and https://github.com/uber-go/zap#quick-start

Comment (nilebox) [reply cmd/smith/app/app.go]: Also I don't think the performance of logging is critical for us. Can't we just always use SugaredLogger?

File: pkg/util/logz/logz.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package logz
```
Comment (nilebox) [inline pkg/util/logz/logz.go]: why `-z` in the package name, tryin' to be cool brah? :)

File: cmd/crd/main.go (hunk 1 above)
Comment (ash2k) [reply cmd/crd/main.go]: No, this is main. What are you going to do if logger initialization fails and you need to print the error?

File: cmd/smith/app/app.go (hunk 2 above)
Comment (ash2k) [reply cmd/smith/app/app.go]: Yes, performance is not critical, so I call `Sugar()` whenever I feel like 😛 
I like that with structured logging (without sugar) I can use helper functions - see `logz` package. How would I do the same with sugared logger?

File: pkg/util/logz/logz.go (hunk 3 above)
Comment (ash2k) [reply pkg/util/logz/logz.go]: log + zap = logz? If you have a better name for it...?

File: cmd/smith/app/app.go (hunk 2 above)
Comment (ash2k) [reply cmd/smith/app/app.go]: Also the sugar call is very cheap, so I don't bother keeping the sugared logger around. That is actually how I started, but then sugared logger does not have the nice typed api.

**Pull request merged**